nextjs/components/EmployerView.tsx

```diff
@@ -21,11 +20,20 @@ function EmployerView() {
   }, [])
 
   const getWorkHoursForDate = (date: Date) => {
-    const workhour = selectedUser?.workHours.find((h) =>
-      areDatesEqual(date, h.startTime)
-    )
-    return workhour?.duration || 0
+    const workhours = selectedUser?.workHours?.map((h) => {
+      if (areDatesEqual(date, h.startTime)) {
+        return h
+      }
+    })
+    if (!workhours || workhours.length === 0) {
+      return 0
+    }
+    let duration = 0
+    workhours.forEach((workhour) => {
```

Comment:
Not really important, but consider a more functional approach, where data isn't mutated. e.g.

```TypeScript
const duration = workhours.reduce((sum, workhour) => sum + workhour?.duration ?? 0, 0)
```

In this case it's no big deal, but in larger code bases where the mutation of the variable occurs far away from the initial assignment, it can be hard to reason about what's happening. A lot of resources out there discussing it:  <https://duckduckgo.com/?q=functional+programming+in+javascript&atb=v310-1&ia=web>